File: ios/LocationSharingApp/LocationSharingApp/Domain/Services/LocationService.swift
import Foundation
import Combine
import CoreLocation

protocol LocationService {
    func startLocationUpdates(interval: TimeInterval)
    func stopLocationUpdates()
    func getCurrentLocation() async -> Location?
    func getLocationUpdates() -> AnyPublisher<Location, Never>
    func requestPermissions() async -> PermissionStatus
    func hasLocationPermission() -> Bool
}

enum PermissionStatus {
    case granted
    case denied
    case shouldShowRationale
    case notRequested
}

File: android/app/src/main/java/com/locationsharing/app/domain/service/LocationService.kt
package com.locationsharing.app.domain.service

import com.locationsharing.app.data.models.Location
import kotlinx.coroutines.flow.Flow

interface LocationService {
    fun startLocationUpdates(interval: Long = 5000)
    fun stopLocationUpdates()
    fun getCurrentLocation(): Location?
    fun getLocationUpdates(): Flow<Location>
    suspend fun requestPermissions(): PermissionStatus
    fun hasLocationPermission(): Boolean
}

enum class PermissionStatus {
    GRANTED, DENIED, SHOULD_SHOW_RATIONALE, NOT_REQUESTED
}